### ISSUE TITLE
CheckBox in ListView multiple select

### ISSUE THREAD

**Issue by sgotliv** [bradmartin/nativescript-checkbox#53]
I have a long ListView with checkboxes. When I randomly check one checkbox in the top of the list and then scroll down I see a few other checkboxes are also have been checked. It happens to me on Android so I believe this is related to view recycling optimization. The issue below was reported a while ago so I hope someone has a workaround for this problem or maybe I am doing something wrong.

https://stackoverflow.com/questions/17856363/checkbox-in-listview-multiple-select

**Comment by bradmartin** [comment]
Sounds like recycling the views. I am guessing at first I would use the `loaded` event on the listview items so that when they are loaded you check if the item has been checked previously and then manually handle the checking/unchecking of the boxes since they're being recycled in the listview. I don't have a sample at the moment but if you make an attempt and need help I'd be happy to talk through it 😄

**Comment by sgotliv** [comment]
@bradmartin Thanks for the quick response! I tried something very similar to your proposal earlier today. I added an additional member "checked" to my model that suppose to keep the state of the relevant checkbox.  The problem is that for some reason the binding between the checkbox and the
model doesn't work, so "recycling" checkbox is checked but I still see checked=false in the model. I am using Angular.

<CheckBox id="{{asset.id}}" [checked]="asset.isChecked (checkedChange)="onCheckBoxToggle(asset.id)"></CheckBox>

**Comment by mmpawar** [comment]
Any update on this? I have the same problem :(

**Comment by kumaran-dena** [comment]
Any fix on this, I also have encountered with the same problem. 

Here my issue description, 

Just assume, the items binding array contains 21 results. If i checked 1st list item in the listview means automatically 9th and 17th, (+7 counts)...etc likewise automatically items are checked. Simultaneously its happening for Uncheck also.

.html
` <ListView *ngIf="searchreslis" [items]="rpersearch" class="list-group">
        <ng-template let-searchResults="item" let-i="index">
          <GridLayout class="list-group-item" rows="*" columns="auto,*,*,*,*">
			<CheckBox col="0" fillColor="#263165" (Tap)="onItemTap(i)"></CheckBox>
            <Label col="1" [text]="searchResults.vim"></Label>
            <Label col="2" [text]="searchResults.vnam"></Label>
            <Label col="3" [text]="searchResults.adt"></Label>
            <Label col="4" [text]="searchResults.asdt"></Label>           
          </GridLayout>
        </ng-template>
      </ListView>`

Any fix/idea on this issue! 
@sgotliv @bradmartin @mmpawar

**Comment by bradmartin** [comment]
I've recently worked on this myself in an app, the key here is to assign unique ids to the checkboxes and handling that in your checkbox itemTap. Since the view is being recycled, you have to maintain the listview items and hook into the loading event and reference the values of your items and handle accordingly.

**Comment by yassern** [comment]
@bradmartin can you show the code, please?

**Comment by bradmartin** [comment]
```xml
 <lv:RadListView.itemTemplate>
                    <StackLayout minHeight="35" class="list-group-item" visibility="{{ name ? 'visible' : 'hidden' }}">
                        <checkbox:CheckBox boxType="square" 
                            id="{{ $value.id + 'serviceCheckbox' }}" 
                            serviceId="{{ $value.id }}" 
                            text="{{ name }}" 
                            tap="{{ onServiceTapped }}" 
                            checked="{{ serviceChecked }}" 
                        	fillColor="#ff517c"
		                    onCheckColor="#fff"
		                    ios:tintColor="#ff517c"
                            class="service-checkbox" />
                        <StackLayout height="100%">
                            <Repeater height="100%" id="{{ id + 'vendorRepeater' }}" row="1" col="0" items="{{ vendors }}" visibility="collapsed">
                                <Repeater.itemsLayout>
                                    <StackLayout width="100%" orientation="vertical" />
                                </Repeater.itemsLayout>
                                <Repeater.itemTemplate>
                                    <checkbox:CheckBox height="30" width="100%" text="{{ name }}" serviceId="{{ $value.serviceId }}" id="{{ $value.serviceId + '_' + $value.id + 'vendorCheckbox' }}" vendorId="{{ $value.vendorId }}" checked="{{ vendorChecked }}" tap="{{ $parents['RadListView'].onServiceVendorTap }}" boxType="circle" class="vendor-checkbox" />
                                </Repeater.itemTemplate>
                            </Repeater>
                        </StackLayout>
                    </StackLayout>
                </lv:RadListView.itemTemplate>
```

```typescript
public onServiceTapped(args) {
		setTimeout(() => {
			const checkbox = args.object as CheckBox;
			const checkboxId = args.object.get("id");
			const serviceId = args.object.get("serviceId");
			const listview = topmost().currentPage.getViewById(
				"servicesListView"
			) as RadListView; // have to get the vm class since we're calling this from a nested view // the current scope is not bound to the vm class `this` instance
			const vendorRepeater = listview.getViewById(
				serviceId + "vendorRepeater"
			) as Repeater;
			const viewModel = listview.bindingContext as ChooseServicesViewModel;
			const serviceItems = viewModel.serviceItems; // get the child repeater for this service
		}, 100);
	}
```

Don't really have time to explain or make this more generic but this is how I work with the ids in listviews bound to items.

**Comment by Atlanta11** [comment]
I work with the baseadapter but my search function don't work with my project and now I test it with RecyclerView and also the same problem.
The rest is working fine.

![knipsel](https://user-images.githubusercontent.com/9640147/50853855-bd5ad180-1383-11e9-9c96-6fa5453ebfc8.PNG)

[https://github.com/Atlanta11/ListView_Multi_Selected_Row_With_Checkbox](url)
